docs/sphinx/source/changelog/pending.rst

```diff
@@ -14,6 +14,9 @@ Testing
 
 Requirements
 ------------
+* Upgrade the notebook environment from python 3.7 to python 3.10.
+  Several dependency versions in ``docs/notebook_requirements.txt`` are
+  updated as well. (:issue:`319`, :pull:`326`)
```

Comment:
Anticipating some additional dependabot PRs before the next release, I thought we might leave this as a placeholder and clean up these requirements.txt change descriptions later.